src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -275,68 +276,48 @@ public Task<ResultSetSubset> GetSubset(int batchIndex, int resultSetIndex, int s
                 return;
             }
 
-            // Open up a connection for querying the database
-            string connectionString = ConnectionService.BuildConnectionString(editorConnection.ConnectionDetails);
-            // TODO: Don't create a new connection every time, see TFS #834978
-            using (DbConnection conn = editorConnection.Factory.CreateSqlConnection(connectionString))
+            // Locate and setup the connection
```

Comment:
I think the issue comes from the fact that this PR used to be for 10 lines of code, but slowly grew to 1000.